redash/authentication/__init__.py

```diff
@@ -110,7 +110,8 @@ def get_user_from_api_key(api_key, query_id):
             if query_id:
                 query = models.Query.get_by_id_and_org(query_id, org)
                 if query and query.api_key == api_key:
-                    user = models.ApiUser(api_key, query.org, query.groups.keys(), name="ApiKey: Query {}".format(query.id))
+                    name = "ApiKey: Query {}".format(query.id)
+                    user = models.ApiUser(api_key, query.org, query.groups.keys(), name, query.user)
```

Comment:
We don't use `ApiUser` for actual users. We load them here:

https://github.com/getredash/redash/blob/3ee7537a6c049a98e7284c67b98130556031e263/redash/authentication/__init__.py#L104

What we need to do is to check if it's a disabled user at this point and return `None` if it is.

Reply:
Could you explain the downsides of my approach? Not that I have a clear understanding of the areas of responsibilities that makes me think it's any better than your suggestion, but just because I lose a bunch of brain cells every time I introduce a null in an app.

Reply:
The main downside is that it's not working 😅... When you use a User's API key, the rest of the system sees this as a regular `User`, not as an `ApiKeyUser`. The later is only used when an API key of a dashboard or query is used...

Reply:
I would call that a winning argument 😆